File: 品味Java的21種設計模式/dp_cpp/bridge/example6/UrgencyMessage.cpp
#include "UrgencyMessage.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace bridge
			{
				namespace example6
				{

					UrgencyMessage::UrgencyMessage(MessageImplementor *impl) : AbstractMessage(impl)
					{
					}

					void UrgencyMessage::sendMessage(std::string message, std::string toUser)
					{
						message = "加急："+message;
						AbstractMessage::sendMessage(message, toUser);
					}

					object *UrgencyMessage::watch(std::string messageId)
					{
						//获取相应的数据，组织成监控的数据对象，然后返回		
						return 0;
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/bridge/example6/UrgencyMessage.java
package cn.javass.dp.bridge.example6;
/**
 * 加急消息
 */
public class UrgencyMessage extends AbstractMessage {
    public UrgencyMessage(MessageImplementor impl) {
        super(impl);
    }
    public void sendMessage(String message, String toUser) {
        message = "加急：" + message;
        super.sendMessage(message, toUser);
    }
    /**
     * 监控某消息的处理过程
     * @param messageId 被监控的消息的编号
     * @return 包含监控到的数据对象，这里示意一下，所以用了Object
     */
    public Object watch(String messageId) {
        //获取相应的数据，组织成监控的数据对象，然后返回
        return null;
    }
}